luaui/Widgets/unit_smart_select.lua

```diff
@@ -207,39 +213,41 @@ function widget:Update()
 
 	local mouseSelection
 
-	if spIsAboveMiniMap(x1, y1) then
+	if inMiniMapSel then
 		mouseSelection = GetUnitsInMinimapRectangle(x1, y1, x2, y2)
 	else
 		mouseSelection = spGetUnitsInScreenRectangle(x1, y1, x2, y2, nil) or {}
 	end
 
-	inSelection = true
-
 	local newSelection = {}
 	local uid, udid, tmp
 
-	-- filter unselectable units
 	tmp = {}
+	local n = 0
+	local equalsMouseSelection = #mouseSelection == #lastMouseSelection
+	local isGodMode = spIsGodModeEnabled()
+
 	for i = 1, #mouseSelection do
 		uid = mouseSelection[i]
-		if not spGetUnitNoSelect(uid) then
-			tmp[#tmp + 1] = uid
+		if not spGetUnitNoSelect(uid) and -- filter unselectable units
+			(not isGodMode and spGetUnitTeam(uid) ~= GaiaTeamID and not ignoreUnits[spGetUnitDefID(uid)] and (spec or spGetUnitTeam(uid) == myTeamID)) then -- filter gaia units + ignored units (objects) + only own units when not spectating
+			n = n + 1
+			tmp[n] = uid
+			if equalsMouseSelection and uid ~= lastMouseSelection[n] then
+				equalsMouseSelection = false
+			end
 		end
 	end
-	mouseSelection = tmp
 
-	-- filter gaia units + ignored units (objects) + only own units when not spectating
-	if not spIsGodModeEnabled() then
-		tmp = {}
-		for i = 1, #mouseSelection do
-			uid = mouseSelection[i]
-			if spGetUnitTeam(uid) ~= GaiaTeamID and not ignoreUnits[spGetUnitDefID(uid)] and (spec or spGetUnitTeam(uid) == myTeamID) then
-				tmp[#tmp + 1] = uid
-			end
-		end
-		mouseSelection = tmp
+	if equalsMouseSelection and mods.idle == lastMods[1] and mods.same == lastMods[2] and mods.deselect == lastMods[3] and mods.all == lastMods[4] and mods.mobile == lastMods[5] then
```

Comment:
If units in selection box is equal to current and modifiers are the same, return early.